zipkin-autoconfigure/storage-elasticsearch-http/src/main/java/zipkin/autoconfigure/storage/elasticsearch/http/ZipkinHttpElasticsearchStorageAutoConfiguration.java

```diff
@@ -25,21 +30,61 @@
 import zipkin.autoconfigure.storage.elasticsearch.ZipkinElasticsearchStorageAutoConfiguration;
 import zipkin.storage.StorageComponent;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 @Configuration
-@EnableConfigurationProperties(ZipkinElasticsearchHttpStorageProperties.class)
+@EnableConfigurationProperties(ZipkinHttpElasticsearchStorageProperties.class)
 @ConditionalOnProperty(name = "zipkin.storage.type", havingValue = "elasticsearch")
-@Conditional(ZipkinHttpElasticsearchStorageAutoConfiguration.HostsAreUrls.class)
+@Conditional({
+    ZipkinHttpElasticsearchStorageAutoConfiguration.HostsAreUrls.class,
+    ZipkinHttpElasticsearchStorageAutoConfiguration.HostsArentAwsUrls.class
+})
 @ConditionalOnMissingBean(StorageComponent.class)
 public class ZipkinHttpElasticsearchStorageAutoConfiguration {
-  @Bean StorageComponent storage(ZipkinElasticsearchHttpStorageProperties elasticsearch) {
+  @Bean StorageComponent storage(ZipkinHttpElasticsearchStorageProperties elasticsearch) {
     return elasticsearch.toBuilder().build();
   }
 
-  public static final class HostsAreUrls implements Condition {
+  static final class HostsAreUrls implements Condition {
     @Override public boolean matches(ConditionContext condition, AnnotatedTypeMetadata md) {
       String hosts = condition.getEnvironment().getProperty("zipkin.storage.elasticsearch.hosts");
       if (hosts == null) return false;
       return ZipkinElasticsearchStorageAutoConfiguration.hostsAreUrls(hosts);
     }
   }
+
+  static final class HostsArentAwsUrls implements Condition {
+    @Override public boolean matches(ConditionContext condition, AnnotatedTypeMetadata md) {
+      String hosts = condition.getEnvironment().getProperty("zipkin.storage.elasticsearch.hosts");
+      if (hosts == null) return true;
+      return !regionFromAwsUrls(Arrays.asList(hosts.split(","))).isPresent();
+    }
+  }
+
+  /**
+   * Only here for consumption by {@link HostsArentAwsUrls}; ideally we could find a way to express
+   * a hierarchy of autoconfigurations in spring without each parent needing to turn themselves off
+   * for their children, but there seems to be no reliable way to order autoconfigure'd classes
+   * (neither @AutoConfigureBefore nor @AutoConfigureAfter seemd to have the desired effect)
+   */
+  private static final Pattern AWS_URL =
+      Pattern.compile("^https://[^.]+\\.([^.]+)\\.es\\.amazonaws\\.com", Pattern.CASE_INSENSITIVE);
+
+  public static Optional<String> regionFromAwsUrls(List<String> hosts) {
```

Comment:
I don't disagree, but this seemed the only way to semi-reliably make a really easy path to get started – setting ES_HOSTS=https://my-aws-domain.us-west-2.amazonaws.com was the first thing I tried, and I think there's value in making that sort of thing work. And since the region is encoded in that URL (and is unfortunately required for the AWS v4 signature stuff), it seemed much more straightforward to peel it out of the semi-standard URL format.

That said, it's definitely the kind of magic that breeds complications. Wether it's worth the tradeoff for ease of getting-started, I suppose, is up to you.